Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs

```diff
@@ -170,8 +170,12 @@ protected override Task<BoolResult> ShutdownCoreAsync(OperationContext context)
                         break;
                     }
 
-                    TimeSpan waitDelay = _retryIntervals[attemptCount];
-                    Tracer.Warning(operationContext, $"{AttemptTracePrefix(attemptCount)} All replicas {hashInfo.Locations.Count} failed. Retrying for hash {hashInfo.ContentHash.ToShortString()}...");
+                    long waitTicks = _retryIntervals[attemptCount].Ticks;
+
+                    // Randomize the wait delay to `[0.5 * delay, 1.5 * delay)`
+                    TimeSpan waitDelay = TimeSpan.FromTicks((long)((waitTicks / 2) + (waitTicks * ThreadSafeRandom.Generator.NextDouble())));
+

```

Comment:
Minor mathy stuff: once you involve floating point (via NextDouble()), probably easiest, safest, and fastest to stay in floating point space. That is: use TimeSpan.FromSeconds as TimeSpan.TotalSeconds. Moving back and forth between integer and floating point is a (tiny) perf cost and, more importantly, often risks weird behaviors such as a failed cast or integer division producing unexpected results (like 1/2 = 0).